classes/webservice.php

```
@@ -766,7 +810,11 @@ public function provide_license($zoomuserid) {
             if ($this->paid_user_limit_reached()) {
                 $leastrecentlyactivepaiduserid = $this->get_least_recently_active_paid_user_id();
                 // Changes least_recently_active_user to a basic user so we can use their license.
-                $this->make_call("users/$leastrecentlyactivepaiduserid", ['type' => ZOOM_USER_TYPE_BASIC], 'patch');
+                if ($leastrecentlyactivepaiduserid) {
+                    $this->make_call("users/$leastrecentlyactivepaiduserid", ['type' => ZOOM_USER_TYPE_BASIC], 'patch');
+                } else {
+                    throw new moodle_exception('errornousersfound', 'mod_zoom');
+                }
```

I agree that it is good to check to match sure that we're making an API call with a good value. However, it doesn't look like this moodle_exception is being caught and it looks like that will prevent the meeting host from being able to join their meeting if there are no licenses available. Where I don't love the idea of failing silently, I do worry about preventing them (and only them) from being able to join the meeting.

---

I added that exception in the first place because when that situation happens (there is no user to borrow the license from) another  exception was already thrown when trying to get the PRO license and there isn't any available.

I agree that throwing an exception is not the best thing to do, so instead of that, I've added the control to try to get the PRO license only if there limit is not reached or if it has been removed from another user, does it make sense to you? Maybe adding a message when this happens could be a good thing to do...